src/controllers/ProductController.ts

```diff
@@ -0,0 +1,35 @@
+import { Request, Response, NextFunction } from 'express';
+import GenericError from '../helpers/GenericError';
+import ProductQuery from 'queries/ProductQuery';
+
+export default class ProductController {
+  static updateProduct = async (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ) => {
+    const id: number = parseInt(req.params.id as string);
```

Comment:
no need, if we extend the Request interface on this request then this type will be defined